source/common/tcp_proxy/tcp_proxy.cc

```diff
@@ -91,7 +91,17 @@ Config::SharedConfig::SharedConfig(
     max_downstream_connection_duration_ = std::chrono::milliseconds(connection_duration);
   }
 
-  if (config.has_access_log_flush_interval()) {
+  if (config.has_access_log_options() &&
+      config.access_log_options().has_access_log_flush_interval()) {
+    if (config.has_access_log_flush_interval()) {
+      throw EnvoyException("Can't use deprecated field 'access_log_flush_interval' together with"
+                           "non-deprecated field TcpAccessLogOptions.access_log_flush_interval.");
```

Comment:
`Only one of access_log_flush_interval from TcpProxy or access_log_flush_interval from TcpAccessLogOptions can be specified.`

Reply:
done